pkg/config/getters.go

```diff
@@ -0,0 +1,143 @@
+	return uint16(viper.GetInt(types.NodeAPIPort))
+}
+
+func APIHost() string {
+	return viper.GetString(types.NodeAPIHost)
+}
+
+func DevstackGetShouldPrintInfo() bool {
+	return os.Getenv("DEVSTACK_PRINT_INFO") != ""
+}
+
+func DevstackSetShouldPrintInfo() {
+	os.Setenv("DEVSTACK_PRINT_INFO", "1")
+}
+
+func DevstackEnvFile() string {
+	return os.Getenv("DEVSTACK_ENV_FILE")
+}
+
+func ShouldKeepStack() bool {
+	return os.Getenv("KEEP_STACK") != ""
+}
+
+type DockerCredentials struct {
+	Username string
+	Password string
+}
+
+func (d *DockerCredentials) IsValid() bool {
+	return d.Username != "" && d.Password != ""
+}
+
+func GetDockerCredentials() DockerCredentials {
+	return DockerCredentials{
+		Username: os.Getenv("DOCKER_USERNAME"),
+		Password: os.Getenv("DOCKER_PASSWORD"),
+	}
+}
+
+func GetLibp2pConfig() (types.Libp2pConfig, error) {
+	var libp2pCfg types.Libp2pConfig
+	if err := ForKey(types.NodeLibp2p, &libp2pCfg); err != nil {
+		return types.Libp2pConfig{}, err
+	}
+	return libp2pCfg, nil
+}
+
+func GetBootstrapPeers() ([]multiaddr.Multiaddr, error) {
+	bootstrappers := viper.GetStringSlice(types.NodeBootstrapAddresses)
+	peers := make([]multiaddr.Multiaddr, 0, len(bootstrappers))
+	for _, peer := range bootstrappers {
+		parsed, err := multiaddr.NewMultiaddr(peer)
+		if err != nil {
+			return nil, err
+		}
+		peers = append(peers, parsed)
+	}
+	return peers, nil
+}
+
+func GetLogMode() logger.LogMode {
+	mode := viper.Get(types.NodeLoggingMode)
+	switch v := mode.(type) {
+	case logger.LogMode:
+		return v
+	case string:
+		out, err := logger.ParseLogMode(v)
+		if err != nil {
+			log.Warn().Err(err).Msgf("invalid logging mode specified: %s", v)
+		}
+		return out
+	default:
+		log.Error().Msgf("unknown logging mode: %v", mode)
+		return logger.LogModeDefault
+	}
+}
+
+func GetLibp2pTracerPath() string {
+	return viper.GetString(types.MetricsLibp2pTracerPath)
+}
+
+func GetEventTracerPath() string {
+	return viper.GetString(types.MetricsEventTracerPath)
+}
+
+func GetExecutorPluginsPath() string {
+	return viper.GetString(types.NodeExecutorPluginPath)
+}
+
+// TODO idk where this goes yet these are mostly random
+
+func GetDownloadURLRequestRetries() int {
+	return viper.GetInt(types.NodeDownloadURLRequestRetries)
+}
+
+func GetDownloadURLRequestTimeout() time.Duration {
+	return viper.GetDuration(types.NodeDownloadURLRequestTimeout)
+}
+
+func SetVolumeSizeRequestTimeout(value time.Duration) {
+	viper.Set(types.NodeVolumeSizeRequestTimeout, value)
+}
+
+func GetVolumeSizeRequestTimeout() time.Duration {
+	return viper.GetDuration(types.NodeVolumeSizeRequestTimeout)
+}
+
+func GetStoragePath() string {
+	// TODO make this use the config when we figure out why our tests fail if this is anything but a tempDir
+	// Currently there is a value in the config file for this but we are ignoring it.
+	storagePath := os.Getenv("BACALHAU_STORAGE_PATH")
+	if storagePath == "" {
+		return os.TempDir()
+	}
+	return storagePath
+}
```

Comment:
For some reason all of our resource-based tests fail if this returns anything but a temp dir. The behavior here is unchanged. When we want this value to be configurable we'll need to debug the breakage changing this value causes.

Reply:
We probably shouldn't be using `BACALHAU_STORAGE_PATH` in tests should we, or does it fail at runtime too?  
Would have thought test specific config would allow us to force this to "" so we get a temp dir?